addons/mail/wizard/mail_compose_message.py

```diff
@@ -591,6 +591,18 @@ def _onchange_template_id(self, template_id, composition_mode, model, res_id):
                 attachment_ids.append(Attachment.create(data_attach).id)
             if values.get('attachment_ids', []) or attachment_ids:
                 values['attachment_ids'] = [Command.set(values.get('attachment_ids', []) + attachment_ids)]
+
+            # Restore default values for fields not updated on template switch
+            fields_to_reset = {'subject', 'body_html', 'email_from', 'email_to', 'partner_to', 'email_cc',  'reply_to', 'mail_server_id'}
+            fields_to_reset -= set(values)
+            if fields_to_reset:
+                values.update(self.default_get(fields_to_reset))
+                # Reset only those inherited from the previous template; preserve values set by e.g. user
+                fields_to_reset_unaltered = fields_to_reset.copy()
+                for field in fields_to_reset:
+                    if field in self and self._origin.template_id[field] != self[field]:
+                        fields_to_reset_unaltered.remove(field)
+                self.with_context(default_composition_mode=composition_mode, default_model=model, default_res_id=res_id).default_get(fields_to_reset_unaltered)
```

Comment:
In composer doesn't have some field -> grab default  (i.e. don't remove this field from the ones that will be set to default)

on a second thought, maybe we should just leave it as it was (eg. if "email_from" wasn't set, leave it unset)